keep-ui/app/settings/add-user-modal.tsx

```diff
@@ -0,0 +1,181 @@
+import { Dialog } from '@headlessui/react';
```

Comment:
Generally, don't know if it's important or not but there's 100% easier (+much less code) way for this screen. Seems like a lot of the states are not needed and some validators can be achieved more easily

Reply:
what's the magic way? and what's states not needed?